easier68k-cli/subcommandline_run.py

```diff
@@ -0,0 +1,225 @@
+        print('sets the value in register to value (decimal unless prefixed with 0x for hexidecimal)')
+        
+    
+    def complete_set_register(self, text, line, begidx, endidx):
+        arg = autocomplete_getarg(line).upper()
+        return [x.name for x in Register if x.name.find(arg) == 0]
+        
+        
+    def do_get_registers(self, args):
+        args = split_args(args, 0, 1)
+        if(args == None):
+            return False
+        
+        if(len(args) == 0):
+            output = ''
+            for i, reg in enumerate(Register):
+                value_hex = long_hex(self.simulator.get_register_value(reg))
+                
+                output += '{:<16}'.format(reg.name + ': ' + value_hex)
+                if(i % 4 == 3):
+                    output += '\n'
+                
+            print(output)
+        else:
+            # get the one passed in
+            try:
+                reg = Register[args[0]]
+                value_hex = long_hex(self.simulator.get_register_value(reg))
+                
+                print(value_hex)
+            except KeyError:
+                print('[ERROR] unrecognized register ' + args[0])
+                return False
+    
+    def complete_get_registers(self, text, line, begidx, endidx):
+        arg = autocomplete_getarg(line).upper()
+        return [x.name for x in Register if x.name.find(arg) == 0]
+        
+        
+    def help_get_registers(self):
+        print('syntax: registers_get [register]')
+        print('if no register is selected then all of them are printed')
+        print('if a register is selected then just that register is printed')
+    
+    
+    
+    def do_dump_memory(self, args):
+        args = split_args(args, 1, 0)
+        if(args == None):
+            return False
+            
+        out_file = open(args[0], 'wb')
+        self.simulator.save_memory(out_file)
+        out_file.close()
+        
+    
+    def help_dump_memory(self, args):
+        print('syntax: dump_memory out_file')
+        print('dumps the memory to the out_file.')
+    
+    def complete_dump_memory(self, text, line, begidx, endidx):
+        return autocomplete_file(line)
+    
+    def do_load_memory(self, args):
+        args = split_args(args, 1, 0)
+        if(args == None):
+            return False
+            
+        try:
+            in_file = open(args[0], 'rb')
+            self.simulator.load_memory(in_file)
+            
+            in_file.close()
+            
+        except FileNotFoundError as not_found:
+            print('[Error] file: ' + str(not_found) + ' does not exist')
+    
+    def help_load_memory(self, args):
+        print('syntax: load_memory in_file')
+        print('loads the memory from the in_file.')
+    
+    def complete_load_memory(self, text, line, begidx, endidx):
+        return autocomplete_file(line)
+        
+    
+    
+    def do_get_memory(self, args):
```

Comment:
could use docstrings, but we are under a time crunch

Reply:
Yup, this can be done in another PR.